### PR TITLE
feat: static getDesignToken

### PR DESCRIPTION
**Pull request by MadCcc**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |      Add static function `getDesignToken` to access full Design Token.     |
| 🇨🇳 Chinese |      新增静态方法 `getDesignToken` 用于获取完整的主题 token。     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at de2e4c8</samp>

This pull request adds a new function `getDesignToken` to the `theme` module, which allows users to access the design token object based on the theme configuration. It also adds type and test support for the function, and updates the documentation in both English and Chinese to reflect the new usage.

### <samp>🤖 Generated by Copilot at de2e4c8</samp>

*  Implement and export `getDesignToken` function to return design token object based on theme configuration ([link](https://github.com/ant-design/ant-design/pull/42723/files?diff=unified&w=0#diff-bfc52d09323f24b5d3cbf6edb9fe8991c8b8fd1daa73c21e4b054bbc69413783R1-R19), [link](https://github.com/ant-design/ant-design/pull/42723/files?diff=unified&w=0#diff-a1691067c7ae7cc4ec252c57ae55fccd18ee928b026091db042495c6cd43e049R32))
*  Import `ThemeConfig` type from `config-provider` context module to use in `getDesignToken` function and test suite ([link](https://github.com/ant-design/ant-design/pull/42723/files?diff=unified&w=0#diff-37ce89d0868e7510a9c83f58a4d4a2cba035535684d5585549df8962f6a7e421R6))
*  Add test suite for `getDesignToken` function using custom render function and `useToken` hook ([link](https://github.com/ant-design/ant-design/pull/42723/files?diff=unified&w=0#diff-37ce89d0868e7510a9c83f58a4d4a2cba035535684d5585549df8962f6a7e421R225-R278))
*  Reorder imports in `theme` module to follow convention ([link](https://github.com/ant-design/ant-design/pull/42723/files?diff=unified&w=0#diff-a1691067c7ae7cc4ec252c57ae55fccd18ee928b026091db042495c6cd43e049L2-R7))
*  Update documentation for `theme` module in English and Chinese versions, replacing `defaultAlgorithm` and `defaultSeed` with `getDesignToken` function and adding example of usage with `ConfigProvider` component ([link](https://github.com/ant-design/ant-design/pull/42723/files?diff=unified&w=0#diff-e12721e5a68f1eaac8d53987399a4d6eb6c7e4e6bb59fed4ffb73c7967079090L163-R199), [link](https://github.com/ant-design/ant-design/pull/42723/files?diff=unified&w=0#diff-7238b66024192aa79bff8e41259edd3a2c30155c7422b58c53e6fc63d679a4afL158-R199))

### CODE REVIEW

File: components/theme/__tests__/token.test.tsx [hunk 1]
```diff
@@ -221,4 +222,58 @@ describe('Theme', () => {
 
     expect(container.querySelector('.duration')?.textContent).toEqual('0s');
   });
+
+  describe('getDesignToken', () => {
+    const getHookToken = (config?: ThemeConfig) => {
```
Comment (kiner-tang) [inline components/theme/__tests__/token.test.tsx:227]: 为啥在这里又实现一遍，不是直接导入方法呢？

File: components/theme/getDesignToken.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { createTheme } from '@ant-design/cssinjs';
+import type { ThemeConfig } from '../config-provider/context';
+import type { AliasToken } from './interface';
+import defaultDerivative from './themes/default';
+import seedToken from './themes/seed';
+import formatToken from './util/alias';
+
+const getDesignToken = (config?: ThemeConfig): AliasToken => {
```
Comment (zombieJ) [inline components/theme/getDesignToken.ts:8]: 先出 RFC？

File: components/theme/__tests__/token.test.tsx (hunk 1 above)
Comment (kiner-tang) [reply components/theme/__tests__/token.test.tsx:227]: 哦，理解错了，跟 getDesignToken 方法不一样

Comment (MadCcc) [reply components/theme/__tests__/token.test.tsx:227]: 这个是用来拿 hook 的结果的

**Comment by li-jia-nan** [comment]
大了200b

File: components/theme/index.ts [hunk 3]
```diff
@@ -28,4 +29,5 @@ export default {
   defaultAlgorithm,
   darkAlgorithm,
   compactAlgorithm,
+  getDesignToken,
```
Comment (li-jia-nan) [inline components/theme/index.ts:32]: 要不要在 `components/index.ts` 里面也导出一下？

**Review by li-jia-nan — APPROVED** [review]

**Pull request merged**